RestaurantReservation.API/Controllers/CustomersController.cs

```diff
@@ -0,0 +1,173 @@
+﻿using Asp.Versioning;
+using AutoMapper;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.JsonPatch;
+using Microsoft.AspNetCore.Mvc;
+using RestaurantReservation.API.DomainErrors;
+using RestaurantReservation.API.Models.Customers;
+using RestaurantReservation.Db.Entities;
+using RestaurantReservation.Db.Repositories;
+using System.Text.Json;
+
+
+namespace RestaurantReservation.API.Controllers
+{
+    /// <summary>
+    /// Managing customer controller
+    /// </summary>
+  
+    [Authorize]
+    [Route("api/customers")]
+    [ApiVersion("1.0")]
+    [ApiController]
+    public class CustomersController : ControllerBase
+    {
+        private readonly CustomerRepository _customerRepository;
+        private readonly IMapper _mapper;
+        private const int DefaultPageSize = 10;
```

Comment:
a note to discuss the proper default page size. also sicne this might be shared among different controllers it's better to add it to a static class or configuration class that we can inject it when we need it.